### PR TITLE
3684

### PR DESCRIPTION
**Pull request by mirterious**
## Номер
3684

## Номер задания
0

## Ссылка на видео с демо
https://www.youtube.com/watch?v=blyjG92h5uE

## Комментарии
Старый ПР
https://github.com/rubizza-camp/summer-2019/pull/194

### CODE REVIEW

File: 3684/0/about_dice_project.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+class DiceSet
+  attr_reader :values
+  def roll(value)
+    @values = []
+    iterator = 0
+    while iterator < value
+      @values << rand(1..6)
+      iterator += 1
+    end
+  end
```
Comment (mesmerze) [inline 3684/0/about_dice_project.rb]: please take a look at http://ruby-doc.org/core-2.6.3/Array.html#method-c-new

as you can see there, a structure like `Array.new(num_of_rolls) { rand(1..6) }` looks much better here) but it's up to you)
just give me your opinion. I'm gonna approve anyway)

File: 3684/0/triangle.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+
+# :reek:FeatureEnvy
+def triangle(aside, bside, cside)
+  array = [aside, bside, cside].sort
+  raise TriangleError if aside <= 0 || bside <= 0 || cside <= 0 || (array[0] + array[1] <= array[2])
```
Comment (mesmerze) [inline 3684/0/triangle.rb]: just 1 micro hint. rename `array` to `triangle_sides`. and user `triangle_sides.first, .second, .third`

how do you think, would it look better?

File: 3684/0/about_dice_project.rb (hunk 1 above)
Comment (mirterious) [reply 3684/0/about_dice_project.rb]: Yep, this is much better. Thanks for helping

File: 3684/0/triangle.rb (hunk 2 above)
Comment (mirterious) [reply 3684/0/triangle.rb]: thats true, its more communicative name in this case

File: 3684/0/triangle.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Triangle Project Code.
+
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/LineLength
```
Comment (mesmerze) [inline 3684/0/triangle.rb]: nobody in your future career gonna like this)

File: 3684/0/triangle.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# The tests for this method can be found in
+#   about_triangle_project.rb
+# and
+#   about_triangle_project_2.rb
+#
+
+# :reek:FeatureEnvy
+def triangle(first, second, third)
+  triangle_sides = [first, second, third].sort
+  raise TriangleError if first <= 0 || second <= 0 || third <= 0 || (triangle_sides[0] + triangle_sides[1] <= triangle_sides[2])
```
Comment (mesmerze) [inline 3684/0/triangle.rb]: triangle_sides[0] -> triangle_sides.first

**Comment by mesmerze** [comment]
@longbeardmen Hi) Seems like my approval is not valid here) Could you please approve or not)

**Pull request merged**